### ISSUE TITLE
Lagging in search when processing over 2000 options

### ISSUE THREAD

**Issue by kingbonio** [pytesNET/tail.select#102]
When using the search box, when typing in the first few characters which matches all options the script locks up and it takes a few seconds for the entered characters to show.

I added a debounce to the search event listener handler which has alleviated this issue but isn't ideal.

```
           this.search.children[0].addEventListener("input", function(event){
                    setTimeout(() => {

                        self.query.call(self, (this.value.length > con.searchMinLength)? this.value: undefined);

                    }, (typeof con.searchDebounce !== undefined) ? con.searchDebounce : 0);

                });

**Comment by SamBrishes** [comment]
Hellow,

thanks for your issue. Version 1.0.0 will contain an activatable / configurable asynchronous search performance with respective indicators for the end-end-user.

Sincerely,
Sam.

**Comment by SamBrishes** [comment]
Finished, get's released in 0.6.0
